control-plane-operator/hostedclusterconfigoperator/controllers/resources/kas/admissionpolicies.go

```diff
@@ -0,0 +1,128 @@
+package kas
+
+import (
+	"context"
+	"fmt"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/hostedclusterconfigoperator/controllers/resources/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/upsert"
+	k8sadmissionv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AdmissionPolicyBlacklist struct {
+	AdmissionPolicies map[string]AdmissionPolicy
+}
+
+type AdmissionPolicy struct {
+	Name        string
+	APIGroup    []string
+	ApiVersion  []string
+	Resources   []string
+	Validations []k8sadmissionv1beta1.Validation
+}
```

Comment:
why do we need those structs and all the helper methods? its kind hard to follow.
it would be good if we keep the same style we use for other resources reconciliation.

Reply:
I've changed how it was structured to be more understandable. Please, let me know if there is any other concern.